modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java

```diff
@@ -115,6 +115,72 @@ public class GridMapQueryExecutor {
     /** */
     private ConcurrentMap<UUID, MapNodeResults> qryRess = new ConcurrentHashMap<>();
 
+
+    /** Default logger initialization */
+    public GridMapQueryExecutor(GridKernalContext ctx) {
+        this.ctx = ctx;
+        this.log = ctx.log(GridMapQueryExecutor.class); 
+    }
+
+    /**
+     * Allows setting a custom logger from outside.
+     * @param log the IgniteLogger to set
+     */
+    public void setLogger(IgniteLogger log) {
+        this.log = log;
+    }
+
+    /**
+     * Example usage of the log in a method
+     */
+    public void executeQuery(String sql) {
+        try {
+            ResultSet rs = executeSql(sql);  // Attempt to execute the SQL.
+            processResults(rs);  // Process the results only if SQL execution succeeds.
+        } catch (SQLException e) {
+            log.error("Failed to execute query: " + sql, e);
+        }
+    }
+    
+    /**
+     * Executes a SQL query against the database.
+     * @param sql SQL statement to execute.
+     * @return ResultSet containing the results.
+     * @throws SQLException if there is a problem executing the SQL.
+     */
+    private ResultSet executeSql(String sql) throws SQLException {
+        if (sql.equals("SELECT * FROM valid_table;")) {
```

Comment:
need to understand whether new methods here are needed.

Reply:
Removed.